src/app/forms/components/weights-trl-hgv/weights-trl-hgv.component.ts

```diff
@@ -0,0 +1,27 @@
+import { Component, Input, OnChanges, OnDestroy, OnInit } from '@angular/core';
+import { DynamicFormService } from '@forms/services/dynamic-form.service';
+import { WeightsComponent } from '../weights/weights.component';
+
+@Component({
+  selector: 'app-weights-trl-hgv[vehicleType]',
+  templateUrl: './weights-trl-hgv.component.html'
+})
+export class WeightsTrlHgvComponent extends WeightsComponent implements OnInit, OnDestroy, OnChanges {
+  @Input() vehicleType!: string;
+
+  constructor(public override dfs: DynamicFormService) {
+    super(dfs);
+  }
+
+  override ngOnInit() {
```

Comment:
Do you need to override these? (genuinely don't know just looks weird)

Reply:
Yeah I agree it looks weird, appears to be a TS ruling, I can turn it off in the TS config but for the sake of a key word I'm okay with it.